src/utils/monitoring.ts

```diff
@@ -290,7 +293,26 @@ export const getBlockDetails = async (api: ApiPromise, blockHash: BlockHash) =>
     records,
     fees.map((fee) => fee.inclusionFee.unwrapOrDefault()),
     feeMultiplier,
-    api.consts.transactionPayment.weightToFee[0]
+    [
+      {
+        coeffInteger: new u128(
+          api.registry,
+          api.runtimeVersion.specName.toString() == "moonbeam" ? 100_000_000_000 : 1_000_000_000
+        ),
+        coeffFrac: api.registry.createType("Perbill", 0),
+        negative: new bool(api.registry, false),
+        degree: new u8(api.registry, 1),
+      },
+      {
+        coeffInteger: new u128(
+          api.registry,
+          api.runtimeVersion.specName.toString() == "moonbeam" ? 100 : 1
+        ),
+        coeffFrac: api.registry.createType("Perbill", 0),
+        negative: new bool(api.registry, false),
+        degree: new u8(api.registry, 3),
+      },
```

Comment:
I see that this parameter is used to calculate `WeightFee` whcih is defined `50 * KiloWei * SupplyFactor` in our runtimes. The specified polynomial here is the length fee, which apparently we are not calculating at all but using from `rpc.payment.queryFeeDetails` - oddly enough both `lenFee` and `baseFee` components there are correct, but not the `adjustedWeightFee` hence the need to compute it.

With the following change, I was able to get accurate fees.



```suggestion
      {
        coeffInteger: new u128(
          api.registry,
          api.runtimeVersion.specName.toString() == "moonbeam" ? 5_000_000 : 50_000
        ),
        coeffFrac: api.registry.createType("Perbill", 0),
        negative: new bool(api.registry, false),
        degree: new u8(api.registry, 1),
      },
```

Reply:
It's defined here https://github.com/PureStake/moonbeam/blob/master/runtime/moonbase/src/lib.rs#L134